spark-jobs/src/main/scala/filodb.downsampler/BatchDownsampler.scala

```diff
@@ -118,7 +119,9 @@ object BatchDownsampler extends StrictLogging with Instance {
     try {
       rawPartsBatch.foreach { rawPart =>
         val rawSchemaId = RecordSchema.schemaID(rawPart.partitionKey, UnsafeUtils.arayOffset)
-        val pkPairs = schemas(rawSchemaId).partKeySchema.toStringPairs(rawPart.partitionKey, UnsafeUtils.arayOffset)
+        val schema = schemas(rawSchemaId)
+        if (schema == Schemas.UnknownSchema) throw UnknownSchemaQueryErr(rawSchemaId)
```

Comment:
If there is an unknown schema, we should log it and move on. Should not fail entire job with the exception.

Reply:
I disagree.  There should never be unknown schema and it is something that needs operator intervention (namely clearing out old keys).  Logging and moving on does not fix the problem, which causes further problems down the line.   At the minimum this needs to be an ERROR in the logs, it really cannot just be ignored.  Also it's not restricted to downsampling, it was discovered during normal querying.

Reply:
I agree that operator intervention may be needed, but why should downsampling be stopped ? How would it benefit ? Would rerun produce a different result ? Lets discuss in person.